### PR TITLE
Problem: datasets do not have uids

### PR DESCRIPTION
**Pull request by mehdisadeghi**
Solution: add uid to datasets

including Particles, Lattice and Mesh.

### CODE REVIEW

**Comment by ahashibon** [comment]
Looks good.

File: simphony/core/keywords.py [hunk 1]
```diff
@@ -1012,6 +1012,13 @@
         shape=[1],
         length=None,
         dtype=None),
+    'CUDS': Keyword(
+        name='CUDS',
+        definition='CUDS Container',  # noqa
```
Comment (stefanoborini) [inline simphony/core/keywords.py]: Please explain better what this is.

Comment (stefanoborini) [inline simphony/core/keywords.py]: In the metadata, of course. This is generated code.

Comment (mehdisadeghi) [inline simphony/core/keywords.py]: Sure. We'll do it in metadata repository.

**Pull request merged**